File: 2021/11/06/main.go
package main

func singleNumber(nums []int) []int {
	var x int
	for _, n := range nums {
		x ^= n
	}

	l := x & -x

	var a, b int
	for _, n := range nums {
		if n&l == 0 {
			a ^= n
		} else {
			b ^= n
		}
	}
	return []int{a, b}
}

File: 2021/11/06/main.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        x = 0
        for n in nums:
            x ^= n

        l, a, b = x & -x, 0, 0

        for n in nums:
            if n & l == 0:
                a ^= n
            else:
                b ^= n
                
        return [a, b]